cscs-checks/libraries/gridtools/gridtools_tests.py

```diff
@@ -9,211 +9,189 @@
                     },
                     'dom:mc': {
-                        'wall_time': (2870, None, 0.1, 'ms')
+                        'timer': (11.0, None, 0.1, 's')
                     },
                     'dom:gpu': {
-                        'wall_time': (3010, None, 0.1, 'ms')
+                        'timer': (1.0, None, 0.1, 's')
                     }
                 }
             },
-            'simple_hori_diff_cpu_kfirst': {
-                'executable_opts': ['100', '100', '100'],
+            'horizontal_diffusion/cpu_ifirst_double': {
+                'executable_opts': [
+                    '256', '256', '80', '1',
+                    '--gtest_filter=horizontal_diffusion/cpu_ifirst_double*'],
                 'reference': {
                     'daint:mc': {
-                        'wall_time': (2880, None, 0.1, 'ms')
+                        'timer': (9.0, None, 0.1, 's')
                     },
                     'daint:gpu': {
-                        'wall_time': (3020, None, 0.1, 'ms')
+                        'timer': (1.0, None, 0.1, 's')
                     },
                     'dom:mc': {
-                        'wall_time': (2880, None, 0.1, 'ms')
+                        'timer': (9.0, None, 0.1, 's')
                     },
                     'dom:gpu': {
-                        'wall_time': (3020, None, 0.1, 'ms')
+                        'timer': (1.0, None, 0.1, 's')
                     }
                 }
-            },
-            'vertical_advection_dycore_gpu': {
-                'executable_opts': ['200', '200', '200'],
+            }
+        }
+
+        self.executable_opts = self.variant_data[variant]['executable_opts']
+        self.reference = self.variant_data[variant]['reference']
+        self.tags = {'scs', 'benchmark'}
+        self.maintainers = ['CB']
+
+    @rfm.require_deps
+    def set_executable(self, GridToolsCPUBuildCheck):
+        self.executable = os.path.join(
+            GridToolsCPUBuildCheck().stagedir,
+            'tests', 'regression', 'perftests')
+
+
+@rfm.parameterized_test(['horizontal_diffusion/gpu_double'],
+                        ['horizontal_diffusion/gpu_horizontal_double'])
+class GridToolsGPURunCheck(GridToolsRunCheck):
+    def __init__(self, variant):
+        super().__init__()
+        # Check if this is a device check
+        self.descr = 'GridTools GPU run test'
+        self.depends_on('GridToolsGPUBuildCheck')
+        self.modules.append('cudatoolkit')
+
+        self.num_gpus_per_node = 1
+
+        self.variant_data = {
+            'horizontal_diffusion/gpu_double': {
+                'executable_opts': [
+                    '256', '256', '80', '1',
+                    '--gtest_filter=horizontal_diffusion/gpu_double*'],
                 'reference': {
                     'daint:gpu': {
-                        'wall_time': (8690, None, 0.1, 'ms')
+                        'timer': (0.0006, None, 0.1, 's')
```

Comment:
How reliable are these timings, because they are rather small. Can you increase the test size to get higher numbers?